### PR TITLE
Raspi tool

### PR DESCRIPTION
**Pull request by Lu-ni**
It's not finished

But we can start to use 
```
make flash-pi
```
and ssh in to use other cmd.

improvement in the future will be:
 * all cmd without ssh
 * faster way to flash
 * remove the power of the emergency stop line
 * cleaning the README
 * a case

### CODE REVIEW

File: tools/raspi_debug_server/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+```BASH
+sudo systemctl list-unit-files | grep enabled
+```
+then
+```BASH
+sudo systemctl start openocd-remote.service
+```
+or 
+```BASH
+reboot
```
Comment (SyrianSpock) [inline tools/raspi_debug_server/README.md]: can also be grouped in one block

```md
## Setup a service (with systemd) to launch our openocd at boot
```BASH
sudo cp openocd-remote.service /lib/systemd/system/
sudo systemctl daemon-reload
sudo systemctl enable openocd-remote.service

# check if it is enabled:
sudo systemctl list-unit-files | grep enabled

# start and reboot
sudo systemctl start openocd-remote.service
reboot
```

File: tools/raspi_debug_server/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+
+systemd services -> sur le raspi pour lancer openocd au démarage
+https://www.tanzolab.it/systemd
+___
+___
+___
+___
+![hey](https://i.imgflip.com/2uhb5u.jpg)
```
Comment (SyrianSpock) [inline tools/raspi_debug_server/README.md]: :wave:

File: tools/raspi_debug_server/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```BASH
+sudo systemctl start openocd-remote.service
+```
+or 
+```BASH
+reboot
+```
+
+
+# note:
```
Comment (SyrianSpock) [inline tools/raspi_debug_server/README.md]: please remove the notes below :eyes:

File: tools/raspi_debug_server/oocd_raspberry.cfg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+telnet_port 4444
+gdb_port 3333
+bindto 0.0.0.0
+# raspberrypi
+interface sysfsgpio
+
+sysfsgpio_swd_nums 25 24
+#nRST = 18
```
Comment (SyrianSpock) [inline tools/raspi_debug_server/oocd_raspberry.cfg]: do we need the reset? or can we remove this line?

File: tools/raspi_debug_server/shell-ssh.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+ssh-copy-id pi@192.168.1.129 && ssh pi@192.168.1.129 -t "python3 -m serial.tools.miniterm /dev/serial/by-id/usb-CVRA_ChibiOS_RT_Virtual_COM_Port_master-if00"
+
+#ssh pi@192.168.1.129 -l pi
```
Comment (SyrianSpock) [inline tools/raspi_debug_server/shell-ssh.sh]: [todo] remove dead code

File: tools/raspi_debug_server/oocd_raspberry.cfg (hunk 4 above)
Comment (Lu-ni) [reply tools/raspi_debug_server/oocd_raspberry.cfg]: We don't but It's wired

File: tools/raspi_debug_server/README.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+```
+```BASH
+sudo systemctl daemon-reload
+```
+```BASH
+sudo systemctl enable openocd-remote.service
+```
+Check if it is enabled:
+```BASH
+sudo systemctl list-unit-files | grep enabled
```
Comment (antoinealb) [inline tools/raspi_debug_server/README.md]: You can also do `systemctl status openocd-remote.service` by the way.

File: tools/raspi_debug_server/oocd_raspberry.cfg (hunk 4 above)
Comment (nuft) [reply tools/raspi_debug_server/oocd_raspberry.cfg]: It's not necessary and can be removed. it was a note in case nRST was needed.

Comment (Lu-ni) [reply tools/raspi_debug_server/oocd_raspberry.cfg]: removed

File: tools/raspi_debug_server/README.md (hunk 6 above)
Comment (Lu-ni) [reply tools/raspi_debug_server/README.md]: I put this one in the README

**Comment by Lu-ni** [comment]
@SyrianSpock  everything done

File: tools/raspi_debug_server/README.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+ * Firmware upload
+ * Acces the USB shell                                          
+ * Connect to the CAN traffic 
+
+Raspberry:
+ * port 3333 : gdb
+ * port 4444 : telnet
+
+ To do:
+ * launch the ./shell-shh.sh script via a cvra cmd
```
[the remark this answers is not in the document]
Comment (SyrianSpock) [reply tools/raspi_debug_server/README.md]: still missing this one :eyes:

**Pull request merged**